Instagram/View Controllers/HomeViewController.m

```
@@ -59,27 +57,49 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     return cell;
 }
 
-- (void) queryPosts {
+- (void) queryPosts:(int) numPosts {
     PFQuery *query = [PFQuery queryWithClassName:@"Post"];
     [query orderByDescending:@"createdAt"];
     [query includeKey:@"author"];
-    query.limit = 20;
+    query.limit = numPosts;
 
     [query findObjectsInBackgroundWithBlock:^(NSArray *posts, NSError *error) {
         if (posts != nil) {
             self.posts = (NSMutableArray *)posts;
             [self.tableView reloadData];
+            self.morePostsLoading = false;
         } else {
             NSLog(@"%@", error.localizedDescription);
         }
     }];
 }
 
 - (void)refreshData:(UIRefreshControl *)refreshControl {
-    [self queryPosts];
+    [self queryPosts:20];
     [refreshControl endRefreshing];
 }
 
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    if(!self.morePostsLoading){
+        int scrollViewContentHeight = self.tableView.contentSize.height;
+        int loadingThreshold = scrollViewContentHeight - self.tableView.bounds.size.height;
+
+        if(scrollView.contentOffset.y > loadingThreshold && self.tableView.isDragging) {
+            self.morePostsLoading = true;
+            PFQuery *query = [PFQuery queryWithClassName:@"Post"];
+            [query countObjectsInBackgroundWithBlock:^(int count, NSError * _Nullable error) {
+                if (!error) {
+                    if (count > self.posts.count) {
+                        [self queryPosts:(self.posts.count+20)];
+                    }
+                } else {
+                    NSLog(@"Error loading more posts");
```

I would fail silently here for dev and user alike.